cmd/kube-apiserver/app/encryption/encryption.go

```diff
@@ -100,17 +123,18 @@ func ParseEncryptionConfiguration(f io.Reader) (map[schema.GroupResource]value.T
 	return result, nil
 }
 
-// GetPrefixTransformer constructs and returns the appropriate prefix transformers for the passed resource using its configuration
-func GetPrefixTransformers(config *ResourceConfig) ([]value.PrefixTransformer, error) {
```

Comment:
@lavalamp If the dependency on gce was the only concern, I think we can move this code back to k8s.io/apiserver.
We don't need any dependency on gce / cloud specific code at all now. We instantiate the cloud in kube-apiserver, and pass a function which returns `cloud.KeyManagementService` as `envelope.Service`, in kube-apiserver. The generic apiserver would not need to know about the cloud.
As for the concern about configuration file format, would it be okay if we parse the config file in kube-apiserver, but using the structs from k8s.io/apiserver? What I mean is, keep the top level config file format struct in kube-apiserver, but move the transformer specific structs to k8s.io/apiserver.

So functions GetPrefixTranformer and the functions which create the transformer will go to k8s.io/apiserver. The only code in kube-apiserver would read the config file, and call `cloud.KeyManagementService`.